server/jetstream_api.go

```diff
@@ -174,10 +178,22 @@ type JSApiStreamInfoResponse struct {
 	*StreamInfo
 }
 
-// JSApiStreamListResponse list of streams.
-type JSApiStreamListResponse struct {
-	Error   *ApiError `json:"error,omitempty"`
-	Streams []string  `json:"streams,omitempty"`
+// Maximum entries we will return.
+// TODO(dlc) - with header or request support could request chunked response.
+const JSApiListLimit = 256
+
+type JSApiStreamsRequest struct {
+	Offset int `json:"offset"`
+}
+
+// JSApiStreamsResponse list of streams.
+// A nil request is valid and means all streams.
+type JSApiStreamsResponse struct {
+	Error   *ApiError     `json:"error,omitempty"`
+	Total   int           `json:"total"`
+	Offset  int           `json:"offset"`
+	Limit   int           `json:"limit"`
+	Streams []*StreamInfo `json:"streams,omitempty"`
```

Comment:
Not sure about this, this is likely to be called a lot - nats utility calls it plenty of times for example - and this is a bunch of information and the paging seems to be only needed because of all the extra information that a basic IsKnownStream() just wouldnt need.

I like this API and want to keep it, but dont like that there is no basic list that just gives me names.

Reply:
Good point on streams and consumers not being omitempty. Will make that change. I can go back to plain names if you feel strongly about it, we will still need paging support however, thing 1M streams, etc..